iocore/eventsystem/ProtectedQueue.cc

```diff
@@ -174,3 +110,14 @@ ProtectedQueue::dequeue_timed(ink_hrtime cur_time, ink_hrtime timeout, bool slee
     }
   }
 }
+
+void
+ProtectedQueue::wait(ink_hrtime timeout)
+{
+  timespec ts = ink_hrtime_to_timespec(timeout);
+  ink_mutex_acquire(&lock);
+  if (INK_ATOMICLIST_EMPTY(al)) {
+    ink_cond_timedwait(&might_have_data, &lock, &ts);
```

Comment:
It is better to keep the code order as the `ProtectedQueue::dequeue_timed`.
Moving `timespec ts = ink_hrtime_to_timespec(timeout);` to here.